src/bitops.c

```diff
@@ -9,6 +9,19 @@
 
 #include "server.h"
 
+#ifdef HAVE_AVX2
+/* Define __MM_MALLOC_H to prevent importing the memory aligned
+ * allocation functions, which we don't use. */
+#define __MM_MALLOC_H
+#include <immintrin.h>
+#endif
+
+#ifdef HAVE_AVX2
+#define BITOP_USE_AVX2 (__builtin_cpu_supports("avx2"))
+#else
+#define BITOP_USE_AVX2 0
+#endif
```

Comment:
use `__builtin_cpu_supports("avx2")` directly below?

Reply:
if `efined(HAVE_AVX2)` is 0, BITOP_USE_AVX2 block won't be compiled either, will it?
Are you concerned that someone will use AVX2 without HAVE_AVX2?

```
#if defined(HAVE_AVX2)
        if (BITOP_USE_AVX2) {
            j = bitopCommandAVX(src, res, op, numkeys, minlen);

            serverAssert(minlen >= j);
            minlen -= j;

            useAVX2 = 1;
        }
#endif
```